pkg/daemon/discover/discover.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package discover to discover unused devices.
+package discover
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"github.com/rook/rook/pkg/util/sys"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	logger          = capnslog.NewPackageLogger("github.com/rook/rook", "rook-discover")
+	AppName         = "rook-discover"
+	NodeAttr        = "rook.io/node"
+	LocalDiskCMData = "devices"
+	LocalDiskCMName = "local-device-"
+)
+
+func Run(context *clusterd.Context) error {
+	if context == nil {
+		return fmt.Errorf("nil context")
+	}
+	nodeName := os.Getenv(k8sutil.NodeNameEnvVar)
```

Comment:
the variables (nodeName and namespace) passed to the daemons can be loaded simply with the cobra command line options. Then you can also check that they are set before the daemon continues. Or if done this way, you may want to return an error if not set